crates/pgt_lsp/src/session.rs

```diff
@@ -446,7 +448,10 @@ impl Session {
                     Ok((vcs_base_path, gitignore_matches)) => {
                         let result = self.workspace.update_settings(UpdateSettingsParams {
                             workspace_directory: self.fs.working_directory(),
-                            configuration: fs_configuration,
+                            configuration: match extra_config {
+                                Some(config) => fs_configuration.clone().merge(config),
+                                None => fs_configuration,
+                            },
```

Comment:
I don't think we need the extra allocation (caused by `clone()`) – plus, if the `extra_config` has any `gitignore` related settings, these would be ignored. 

Maybe it is better to do a `fs_configuration.merge_with(extra_config)` a few lines above (if extra_config is `Some(..)`)? 

If you use the `.merge_with` method from the biome Merge trait, you probably don't even need an extra method 👍

Let me know if you need help with that.